File: wave2_network/packages/shared-py/blindference_utils/commitment.py
from __future__ import annotations

import hashlib


def hash_output(text: str) -> bytes:
    return hashlib.sha256(text.encode("utf-8")).digest()


def build_commitment_hash(output_cid: str, output_hash: bytes) -> str:
    return hashlib.sha256(output_cid.encode("utf-8") + output_hash).hexdigest()

File: wave2_network/packages/shared/src/commitment.ts
import crypto from 'crypto'

export function hashOutput(text: string): Buffer {
  return crypto.createHash('sha256').update(text, 'utf8').digest()
}

export function buildCommitmentHash(outputCID: string, outputHash: Buffer): string {
  return crypto.createHash('sha256').update(Buffer.concat([Buffer.from(outputCID, 'utf8'), outputHash])).digest('hex')
}
